src/validators/json.rs

```diff
@@ -87,7 +89,7 @@ impl Validator for JsonValidator {
 pub fn validate_json_bytes<'a, 'py>(
     input: &'a (impl Input<'py> + ?Sized),
 ) -> ValResult<ValidationMatch<EitherBytes<'a, 'py>>> {
-    match input.validate_bytes(false) {
+    match input.validate_bytes(false, ValBytesMode { ser: BytesMode::Utf8 }) {
```

Comment:
Could we use a default `mode` in the `validate_bytes` function? Where's the equivalent code for `serialize_bytes`, and do we use a default there?

Reply:
I think this code is reusing the *field validator* for a second purpose: converting `validate_json`'s argument to Rust-usable bytes, because the caller can pass different types (`str`, `bytes`, `bytearray`, or any subclasses). pydantic-core only accepts "normal" bytes as the top level `validate_json` argument (no hex or base64 encoding there).

There isn't really an analogous thing for serialization because `to_json` always emits `bytes`, with no alternative output types.

(I am new to all of this so apologies if I'm misunderstanding.)